src/dataflow/core/lispress.py

```diff
@@ -527,6 +547,32 @@ def unnest_line(
                 )
                 result_exprs.extend(exprs)
             return result_exprs, arg_idx, idx, var_id_bindings
+
+        elif hd == DataflowFn.Lambda.value:
+            # (lambda (^ArgType arg_name) body)
+            assert (
+                len(tl) == 2 and len(tl[0]) == 1
+            ), f"{DataflowFn.Lambda.value} binding must have a single arg, and a single body"
+            (arg,), body = tl
+            assert (
+                len(arg) == 3 and arg[0] == META_CHAR
+            ), f"{DataflowFn.Lambda.value} arg must have a type ascription, and a name"
```

Comment:
Does it really have to have a type ascription?

Reply:
Fine to leave a TODO if it doesn't, and I think it doesn't really?

Reply:
I don't think it does but maybe we always include it in the v2 data (?).

Reply:
Pretty sure it never appears without a type ascription in the data (I verified that all of TreeDST can be round-tripped w/o Exceptions, and i think there are no lambdas in calflow). If it's not a real requirement I can change the code to work without it.